profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry_mock/src/Mock/PoetryMock.php

```diff
@@ -458,25 +467,45 @@ private static function getDemandeIdFromJobReference($job_reference) {
    *
    * @param string $job_reference
    *   A tmgmt_poetry job reference.
+   *
+   * @return array
+   *   An array with reference and request file details.
    */
   public static function getTranslationRequestByJobReference($job_reference) {
     $demande_id = self::getDemandeIdFromJobReference($job_reference);
 
-    $file_path = TMGMT_POETRY_MOCK_REQUESTS_PATH . implode('_', $demande_id) . '.xml';
+    $file_name = self::prepareReferenceNumber($demande_id) . '.xml';
 
-    $result = db_select('file_managed', 'fm')
-      ->fields('fm', ['fid'])
-      ->condition('filemime', 'application/xml', '=')
-      ->condition('uri', $file_path)
-      ->execute()
-      ->fetchAllAssoc('fid');
+    $query = new \EntityFieldQuery();
+    $query->entityCondition('entity_type', 'file')
+      ->propertyCondition('filename', $file_name);
+    $result = $query->execute();
 
     if ($result) {
+      $file_info = reset($result['file']);
       return array(
         'demande_id' => $demande_id,
-        'file' => file_load(key($result)),
+        'file' => file_load($file_info->fid),
       );
     }
   }
 
+  /**
+   * Returning properly formatted reference number string.
+   *
+   * @param array $demande_id
+   *   An array with reference elements.
+   *
+   * @return string
+   *   The string with properly formatted reference number.
+   */
+  public static function prepareReferenceNumber($demande_id) {
```

Comment:
Mock is mimicking Poetry Web service so it should have own functions as it is a server. Embedding client functions inside mocked server is binding us a bit. Here we have more flexibility.